### ISSUE TITLE
saveFrame() doesn't save opaque PNG files.

### ISSUE THREAD

**Issue by vormplus** [processing/processing#3031]
Documentation for `saveFrame()` says: All images saved from the main drawing window will be opaque.

However, if I try to save PNG files when only calling `background()` in `setup()`, the output is a transparent PNG with faded colors. If I open the PNG file in Photoshop and put a white background behind it, I still don't get the vibrant colors I see when I run the sketch.

Top of the screenshot shows the Processing window, bottom shows the saved PNG image.
![screen shot 2015-01-06 at 10 06 51](https://cloud.githubusercontent.com/assets/805141/5626745/d6ab2ebc-958f-11e4-9675-7132b83a14e9.png)

I'm using Processing 2.2.1 on Mac OS X 10.9.5. Operating system was installed from scratch last week.

Here's the code for the sketch:

	color[] colors = { #479487, #9CC3BD, #F1EBDC, #FFC517, #FFAA00 };
	
	boolean saveOne = false;
	
	int r;
	
	void setup()
	{
		size( 1920, 1200, P3D );
		smooth( 8 );
		background( 255 );
	
		r = floor( random( 10000, 90000 ) );
	}
	
	void draw()
	{
		color s, f;
	
		if ( random( 100 ) < 10 ) {
			s = #B3031B;
			f = #B3031B;
		} else {
			s = colors[ floor( random( colors.length ) ) ];
			f = colors[ floor( random( colors.length ) ) ];
		}
	
		stroke( s, 255 );
		fill( f, 64 );
		
		pushMatrix();
		translate( random( width ), random( height ) );
		rotate( random( TWO_PI ) );
		if ( random( 100 ) < 10 ) {
			box( random( 60, 120 ) );
		} else {
			box( random( 10, 40 ) );
		}
		popMatrix();
		
		if ( saveOne || frameCount % 2000 == 0  ) {
			saveFrame("images/yellowcubes-" + r + "-#####.png");
			saveOne = false;
		}
	}
	
	void keyPressed()
	{
		if ( key == 's' ) {
			saveOne = true;
		}
	}

**Comment by toniholzer** [comment]
To save images with alpha channel would be a helpful improvement !!!
If it's not possible, PNG's should at least be saved like the original looks like.

**Comment by GKFX** [comment]
Confirmed on Ubuntu 14.04. The same applies in P2D, so it's an OpenGL issue. If the fill color is set to opaque, the problem goes away. More interestingly, if you take the image and set the alpha value of all pixels to opaque, the image is fully restored. (Using ImageMagick's `compare` between a restored file and a screenshot showed no difference.)
![restored](https://cloud.githubusercontent.com/assets/5357642/5780623/ff6d6704-9da3-11e4-9bdd-461754ad8f40.png)

**Comment by GKFX** [comment]
@toniholzer You just work in your own PGrahpics from createGraphics() to get an alpha channel.

**Comment by benfry** [comment]
Have you tried with 3.0a5?

**Comment by hamoid** [comment]
I tried with 2.2.1 and 3.0a5 on 64 bit Arch Linux. 
For me the issue is only present on 2.2.1. Works fine on 3.0a5.

**Comment by benfry** [comment]
Thanks for checking, @hamoid.

**Comment by JakubValtar** [comment]
This issue is only masked in 3.0a5. Alpha is now additive, so you are saving opaque image.

To expose the issue, run this:
```processing
size(200, 200, P2D);
background(255, 10);
saveFrame("frame.png");
exit();
```

I discussed with @codeanticode how to solve this and it should be fixed in #3067.
